### PR TITLE
[Issue #3712][python-client] exposing InitialPosition management in the ConsumerConfiguration.

### LINKED ISSUE #3712: Be able to manage ConsumerConfiguration SubscriptionInitialPosition with the Python client.
**Is your feature request related to a problem? Please describe.**
SubscriptionInitialPosition have been added in PR #3494.
However, it is still impossible to manage it when creating a Consumer with the python client.

**Describe the solution you'd like**
From what I saw, all the code exist already in the CPP client.

The only missing thing is that some methods are not exposed yet in the ConsumerConfiguration.
- `ConsumerConfiguration::setSubscriptionInitialPosition`
- `ConsumerConfiguration::getSubscriptionInitialPosition`

The object `InitialPosition` also have to be exposed.

Once these methods exposed, adding a parameter to [Client.subscribe()](https://github.com/apache/pulsar/blob/1f376e10362df30af35580f072e83d57757013d2/pulsar-client-cpp/python/pulsar/__init__.py#L464) should be fine.

### PR DESCRIPTION
**Pull request by lelabo-marc**
Fixes #3712

### Motivation
PR #3567 introduced the SubscriptionInitialPosition option in ConsumerConfiguration in the CPP client but did not exposed any methods to be able to do it with the Python client.

This PR aims to expose the code introduced in the previous PR to allows Python user to choose the initial position of the consumer in python. 

### Modifications

- Implemented a boost object to expose InitialPosition Enum.
- Added to boost ConsumerConfiguration object the getter/setter of the InitialPosition attribute.
- Added a initial_position parameter to Client.subscribe in order to modify the ConsumerConfiguration instance created.

### Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

### Does this pull request potentially affect one of the following parts:

  - Anything that affects deployment: (don't know but it should not)

### Documentation

This PR introduce a new feature for the Python client.
I added a small description of how to use it by modifying the pulsar/__init__.py as I think the sources are used to generate the documentation.

### CODE REVIEW

**Comment by lelabo-marc** [comment]
Hi @merlimat, can you please take a look ?
Did my best for the tests but did not achieve to run the docker scripts.
However, I compilled the source and runned a few tests on a remote standalone without crossing any errors.
I don't think they should be many side effects as I just exposed code from the CPP client using boost and did not changed the default behaviour of the consumer.

File: pulsar-client-cpp/python/pulsar/__init__.py [hunk 1]
```diff
@@ -536,6 +537,10 @@ def my_listener(consumer, message):
           can be used for identify a consumer at broker side.
         * `pattern_auto_discovery_period`:
           Periods of seconds for consumer to auto discover match topics.
+        * `initial_position`:
+          Set the initial position of a consumer  when subscribing to the topic.
+          It could be either: `InitialPosition.InitialPositionEarliest` or `InitialPosition.InitialPositionEarliest`.
```
Comment (merlimat) [inline pulsar-client-cpp/python/pulsar/__init__.py]: `Earliest` or `Latest`

File: pulsar-client-cpp/python/pulsar/__init__.py [hunk 2]
```diff
@@ -472,7 +472,8 @@ def subscribe(self, topic, subscription_name,
                   broker_consumer_stats_cache_time_ms=30000,
                   is_read_compacted=False,
                   properties=None,
-                  pattern_auto_discovery_period=60
+                  pattern_auto_discovery_period=60,
+                  initial_position=None
```
Comment (merlimat) [inline pulsar-client-cpp/python/pulsar/__init__.py]: The default here is `Latest` anywyay, so we could set it to that instead of None

**Comment by merlimat** [comment]
@lelabo-marc You can run the test by spinning up a test standalone cluster:  `pulsar-client-cpp/pulsar-test-service-start.sh` 

Then `cd pulsar-client-cpp/python` and `python pulsar_test.py`

**Comment by lelabo-marc** [comment]
Followed your recommendation and implemented a new test. I also corrected the doc and shortened the name of the enum parameters.

However, 3 tests are currently failing but they seems to be not related to the current PR. The python tests are successful.

**Comment by merlimat** [comment]
run integration tests

**Comment by merlimat** [comment]
run java8 tests

**Comment by merlimat** [comment]
run java8 tests
run integration tests

**Comment by sijie** [comment]
run java8 tests

**Comment by sijie** [comment]
run java8 tests

**Comment by sijie** [comment]
run cpp tests

**Comment by merlimat** [comment]
Merged in 2.3.1 at 
c078e71

**Pull request merged**